### PR TITLE
remove usage of 256 as magical number for the height of a chunk

### PR DESCRIPTION
**Pull request by KillingSpark**
I replaced all appearances, that i found, of numbers with cChunkDef::Height, where it made sense to me. Maybe I didn't catch everything in the generating folder since I'm not all to familiar with the generation-process.

src\Blocks\blockleaves.h(169) 255 -> height - 1
src\Entities\Entity.cpp(1461): 256 -> height
src\Entities\Entity.cpp(1493): 128 -> height / 2
src\Generating\FinishGen.cpp(520): 255 -> height - 1
src\Generating\FinishGen.cpp(1224): 255 -> height - 1
src\Defines.h(560): 255 -> height - 1
src\WorldStorage\MapSerializer.cpp(166): 256 -> height
src\WorldStorage\SchematicFileSerializer.cpp(167): 256 -> height
src\ClientHandle.cpp(80): 256 -> height + 1 for a invalid Y
src\ClientHandle.cpp(84): 256 -> height + 1 for a invalid Y

In hope it's of any help,
Spark

### CODE REVIEW

**Comment by KillingSpark** [comment]
I chose height + 1, so it is obvious for anyone that the value is invalid. Height would have done the same but that way it is much clearer

File: src/ClientHandle.cpp [hunk 1]
```diff
@@ -77,11 +77,11 @@ cClientHandle::cClientHandle(const AString & a_IPString, int a_ViewDistance) :
 	m_BlockDigAnimStage(-1),
 	m_BlockDigAnimSpeed(0),
 	m_BlockDigAnimX(0),
-	m_BlockDigAnimY(256),  // Invalid Y, so that the coords don't get picked up
+	m_BlockDigAnimY(cChunkDef::Height + 1),  // Invalid Y, so that the coords don't get picked up
```
Comment (sphinxc0re) [inline src/ClientHandle.cpp]: So you are checking for 257? What is the exact value of `cChunkDef::Height`?

File: src/ClientHandle.cpp [hunk 2]
```diff
@@ -77,11 +77,11 @@ cClientHandle::cClientHandle(const AString & a_IPString, int a_ViewDistance) :
...
 	m_BlockDigAnimZ(0),
 	m_HasStartedDigging(false),
 	m_LastDigBlockX(0),
-	m_LastDigBlockY(256),  // Invalid Y, so that the coords don't get picked up
+	m_LastDigBlockY(cChunkDef::Height + 1),  // Invalid Y, so that the coords don't get picked up
```
Comment (sphinxc0re) [inline src/ClientHandle.cpp]: Same here

File: src/ClientHandle.cpp (hunk 1 above)
Comment (KillingSpark) [inline src/ClientHandle.cpp]: the exact value of cChunkDef::Height is 256. So yes im checking for 257. I did it so it's really clear, that this is an invalid value, which was the purpose of the "256" in the original version. Just cChunkDef::Height would have been enough.

Comment (madmaxoft) [inline src/ClientHandle.cpp]: @sphinxc0re The actual value assigned doesn't make much difference, as long as it's an invalid Y coord. The purpose of this variable triplet is to de-duplicate and match dig requests, so there's not even an explicit check for validity, only for "equals the last set sent by the client". I support this change, as it makes the value obviously invalid.

**Comment by madmaxoft** [comment]
Please don't merge master to your branch. Rebase instead (next time).

**Comment by KillingSpark** [comment]
You mean i should use an other branch then master in my fork? I dont really get what you mean sorry

**Comment by madmaxoft** [comment]
I assume you used the "Update branch" button on this page? Don't do that. It creates tangled history, we prefer to rebase. If you use command-line git, this is what you should do:
```
git remote add upstream https://github.com/cuberite/cuberite.git
git fetch upstream
git rebase upstream/master
```
The first line needs to be executed only once, it will bind your fork to the main repo and allow you to fetch changes in the main repo. The second line fetches the changes. The third line rebases your changes so that they appear as if made on top of the current main repo. If the main repo changes again, you can execute the second and third line again to get to the newest version.

Occasionally, the rebasing will produce a conflict (if someone else edited the same part of the source as your changes), you will need to resolve that conflict and continue with the merge. Or feel free to ask for help :)

**Comment by KillingSpark** [comment]
Thanks for the explanation. I'm still new to Git(hub) so I appreciate it very much :)

**Comment by madmaxoft** [comment]
As I said, don't worry about it for this PR. Just next time.

**Comment by madmaxoft** [comment]
Please stop the merging!

**Comment by madmaxoft** [comment]
I'll manually cherry-pick the changes. Merging would most likely be a disaster for the history.

**Comment by madmaxoft** [comment]
Merged manually.

**Pull request closed without merging**